// [[Reason: Cannot add property to class expressions]] ////

//// [tests/cases/conformance/salsa/typeFromPropertyAssignment29.ts] ////

===================================================================
--- TSC	declarations
+++ DTE	declarations
@@ -55,13 +55,9 @@
 declare class ExpandoClass {
     n: number;
 }
 declare var n: number;
-declare var ExpandoExpr3: {
-    new (): {
-        n: number;
-    };
-};
+declare var ExpandoExpr3: invalid;
 declare var n: number;
 //# sourceMappingURL=typeFromPropertyAssignment29.d.ts.map
 /// [Errors] ////
 
@@ -74,15 +70,16 @@
 typeFromPropertyAssignment29.ts(100,14): error TS2339: Property 'prop' does not exist on type 'typeof ExpandoClass'.
 typeFromPropertyAssignment29.ts(101,14): error TS2339: Property 'm' does not exist on type 'typeof ExpandoClass'.
 typeFromPropertyAssignment29.ts(104,22): error TS2339: Property 'prop' does not exist on type 'typeof ExpandoClass'.
 typeFromPropertyAssignment29.ts(104,42): error TS2339: Property 'm' does not exist on type 'typeof ExpandoClass'.
+typeFromPropertyAssignment29.ts(107,20): error TS9022: Inference from class expressions is not supported with --isolatedDeclarations.
 typeFromPropertyAssignment29.ts(110,14): error TS2339: Property 'prop' does not exist on type 'typeof ExpandoExpr3'.
 typeFromPropertyAssignment29.ts(111,14): error TS2339: Property 'm' does not exist on type 'typeof ExpandoExpr3'.
 typeFromPropertyAssignment29.ts(114,22): error TS2339: Property 'prop' does not exist on type 'typeof ExpandoExpr3'.
 typeFromPropertyAssignment29.ts(114,42): error TS2339: Property 'm' does not exist on type 'typeof ExpandoExpr3'.
 
 
-==== typeFromPropertyAssignment29.ts (14 errors) ====
+==== typeFromPropertyAssignment29.ts (15 errors) ====
     function ExpandoDecl(n: number): string {
         return n.toString();
     }
     declare namespace ExpandoDecl {
@@ -208,8 +205,10 @@
 !!! error TS2339: Property 'm' does not exist on type 'typeof ExpandoClass'.
     
     // Class expressions shouldn't work in typescript either
     var ExpandoExpr3 = class {
+                       ~~~~~
+!!! error TS9022: Inference from class expressions is not supported with --isolatedDeclarations.
         n = 10001;
     }
     ExpandoExpr3.prop = 3
                  ~~~~
